spec/factories/allocations.rb

```diff
@@ -16,5 +17,15 @@
     trait :initial do
       request_type { "initial" }
     end
+
+    trait :with_allocation_uplift do
+      association :allocation_uplift
+    end
+
+    after :build do |allocation|
+      # Neccessary hack to get attributes to persist through JSONAPI build
```

Comment:
Can you elaborate? Not currently clear to me what this is for. Looks like `FactoryBot.build(:allocation, :with_allocation_uplift)` builds the objects I would expect with the correct data?

Reply:
Suggest we add a link to the issue in the comment. This should all be quite temporary as we move to merge Publish/TTAPI, but worth clarifying any potential confusion.

Can this `after :build` be nested under the `trait :with_uplift`, so that it only occurs when that trait is used?

Reply:
Agreed, added both suggestions 👍